packages/curve-ui-kit/src/widgets/Disclaimer/HeaderPage.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react'
+import { useRouter } from 'next/router'
+import { t } from '@lingui/macro'
+
+import Stack from '@mui/material/Stack'
+import IconButton from '@mui/material/IconButton'
+import Typography from '@mui/material/Typography'
+import WestIcon from '@mui/icons-material/West'
+
+import { SizesAndSpaces } from 'curve-ui-kit/src/themes/design/1_sizes_spaces'
+import { RiskDisclaimersIcon } from 'curve-ui-kit/src/shared/icons/RiskDisclaimersIcon'
+
+const { Spacing, Sizing, MaxWidth } = SizesAndSpaces
+
+export const HeaderPage = () => {
+  const router = useRouter()
+
+  const [canGoBack, setCanGoBack] = useState(false)
+  // Check if window.history has previous entries from our app
+  // TODO: this doesn't work properly yet.
+  useEffect(() => {
+    setCanGoBack(window.history.length > 1 && document.referrer.includes(window.location.host))
+  }, [])
```

Comment:
useMemo is simpler here
```suggestion
  const canGoBack = useMemo(() => window.history.length > 1 && document.referrer.includes(window.location.host), [])
```